embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/telemetry/EmbraceTelemetryService.kt

```diff
@@ -0,0 +1,77 @@
+package io.embrace.android.embracesdk.telemetry
+
+import io.embrace.android.embracesdk.capture.metadata.EmbraceMetadataService
+import io.embrace.android.embracesdk.internal.spans.toEmbraceAttributeName
+import java.util.Locale
+
+/*
+    Service for tracking usage of public APIs, and different internal metrics about the app.
+ */
+internal class EmbraceTelemetryService(
```

Comment:
Worth chatting to @bidetofevil about the best approach for cleaning data as he's more familiar with the span session model, but I think it'd be worth making this implement `DataCaptureService`. That interface has functions that handle getting capturing data & also cleaning old data when the session boundary ends.

Reply:
The issue I have with the existing `DataCaptureService` model is the lack of atomicity for flushing and clearing. It maybe less important for general telemetry (although I think this is something we should improve on as well), but I'd like to see an iteration on that where we can enforce atomicity with like a `flush()` function.

If this works right now, I would take a look at that later, perhaps as an extension of the interface or another that builds on it, and make this adhere to that.